### PR TITLE
bug/1995 clamav not detecting malware

### LINKED ISSUE #1995: As tech lead, I need clamav to detect test virus files
Thank you for taking the time to let us know about the issue you found. The basic rule for bug reporting is that 
something isn't working the way one would expect it to work. Please provide us with the information requested 
below and we will look at it as soon as we are able.

## Description

I submitted test virus files in dev-qasp and staging environments and clamav did not detect these as infected. 

## Action Taken

As an `OFA Admin`, I submitted [test infected files](https://en.wikipedia.org/wiki/EICAR_test_file) on the [qasp frontend data files page](https://tdp-frontend-qasp.app.cloud.gov/data-files) and the [staging frontend data files pages](https://tdp-frontend-staging.app.cloud.gov/data-files) 

## What I expected to see

The video [here](https://github.com/raft-tech/TANF-app/issues/1215) demonstrates the expected behavior. 

## What I did see

- successful submission notification on frontend 
- successful submission on backend
clamav0
- "clean" scan result on backend
clamav1
- evidence that test file is otherwise detected as malicious
clamav2

## Other Helpful Information
+ [infected file test script in our codebase](https://github.com/raft-tech/TANF-app/blob/f5d931eb266471aa6e758142d05071f39d8451da/tdrs-backend/tdpservice/test/test_clamav.py#L37)
+ URL of the page I was on: see description
+ Browser and version: chrome
+ Operating System: mac
+ Is the issue repeatable?:yes
+ Has the issue occurred more than once?: yes

### PR DESCRIPTION
**Pull request by n0remac**
## Summary of Changes
Pull request closes #1995 

**Testing:**

1. Deploy to an environment.
2. Log in and go to Data File.
3. Attempt to upload an eicar test file.
4. There should be an error.
5. The clamav admin page should report that malware was detected.

I removed the AV_SCAN_URL env variable from all environments. This value does not need to be modified by the build pipeline and can be set in the Django settings within the code base.

### CODE REVIEW

**Review by ADPennington — COMMENTED** [review]
@n0remac I tested this in [dev-raft](https://tdp-frontend-raft.app.cloud.gov/) environment to confirm what happens to the value for AV_SCAN_URL after deploying the changes from this branch. 

Before deploy:
`AV_SCAN_URL: http://tanf-dev-clamav-rest.apps.internal:9000`

After [deploy](https://app.circleci.com/pipelines/github/raft-tech/TANF-app/9562/workflows/227cea7c-d928-4810-9ad2-0adc3d9e2831/jobs/23866/parallel-runs/0/steps/0-106):
`AV_SCAN_URL: http://tanf-dev-clamav-rest.apps.internal:9000`

Is it possible that CircleCI needs a correct version of this key-value pair in order to overwrite pre-existing values in cloud.gov? I noticed in the deploy backend step, av scan does not get set despite it being on the list in the shell script. I'm guessing this is because the value isnt empty, but i'm not sure where the script is looking for the value.

**Comment by n0remac** [comment]
I am going to record my tests here:
I have set AV_SCAN_URL to an incorrect value. I am going to deploy to qasp and I expect that the value will not get overwritten because I haven't fixed anything yet.
Screen Shot 2022-07-29 at 11 20 30 AM

**Comment by n0remac** [comment]
I hardcoded the values for staging, dev, and prod. I am going to test it again to see if the variable changes. I think it did not work last time because there was an existing value so the default one was not used.

**Comment by n0remac** [comment]
The app no longer uses environment variables to set the clamav endpoint value. Instead they are hard coded in cloudgov.py: 
Screen Shot 2022-07-29 at 1 31 48 PM

then it is referenced from the django settings:
Screen Shot 2022-07-29 at 1 29 33 PM


Because of that we see that the environment variable has not changed but the app still works as expected because it is getting the correct variable from the hardcoded values.

Screen Shot 2022-07-29 at 1 21 57 PM

I do not believe we need to set the variable from circleci. If there concerns about making this value publicly known in the code we can change it. But this should be fine.

File: tdrs-backend/tdpservice/settings/cloudgov.py [hunk 1]
```diff
@@ -125,7 +125,7 @@ class Development(CloudGov):
 
     # https://docs.djangoproject.com/en/2.0/ref/settings/#allowed-hosts
     ALLOWED_HOSTS = ['.app.cloud.gov']
-    AV_SCAN_URL = os.getenv('AV_SCAN_URL', 'http://tanf-dev-clamav-rest.apps.internal:9000')
+    AV_SCAN_URL = 'http://tanf-dev-clamav-rest.apps.internal:9000/scan'
```
Comment (n0remac) [inline tdrs-backend/tdpservice/settings/cloudgov.py:128]: inherit from absract

**Review by ADPennington — APPROVED** [review]
looks good @n0remac 🚀 test notes below:

- AV_SCAN_URL deleted from cloud.gov and redeployed to qasp env via circleci
- good file:
```
11:08:45.584: [APP/PROC/WEB.0] [2022-08-04 15:08:45 +0000] [566] [DEBUG] POST /v1/data_files/
11:08:45.613: [APP/PROC/WEB.0] [2022-08-04 15:08:45,612 DEBUG validators.py::validate_file_infection:L54 :  CLAMAV_NEEDED noted as True, proceeding with scan.
11:08:45.613: [APP/PROC/WEB.0] CLAMAV_NEEDED noted as True, proceeding with scan.
11:08:45.613: [APP/PROC/WEB.0] [2022-08-04 15:08:45,612 DEBUG clients.py::__init__:L37 :  Set clamav endpoint_url as 'http://tanf-dev-clamav-rest.apps.internal:9000/scan'
11:08:45.613: [APP/PROC/WEB.0] Set clamav endpoint_url as 'http://tanf-dev-clamav-rest.apps.internal:9000/scan'
11:08:45.613: [APP/PROC/WEB.0] [2022-08-04 15:08:45,613 DEBUG clients.py::scan_file:L64 :  Initiating virus scan for file: test_Section1
11:08:45.613: [APP/PROC/WEB.0] Initiating virus scan for file: test_Section1
11:08:45.755: [APP/PROC/WEB.0] [2022-08-04 15:08:45,754 DEBUG clients.py::scan_file:L90 :  File scan marked as CLEAN for file: test_Section1
11:08:45.755: [APP/PROC/WEB.0] File scan marked as CLEAN for file: test_Section1

```
- bad file:
```
1:11:05.146: [APP/PROC/WEB.0] [2022-08-04 15:11:05 +0000] [566] [DEBUG] POST /v1/data_files/
11:11:05.177: [APP/PROC/WEB.0] [2022-08-04 15:11:05,176 DEBUG validators.py::validate_file_infection:L54 :  CLAMAV_NEEDED noted as True, proceeding with scan.
11:11:05.177: [APP/PROC/WEB.0] CLAMAV_NEEDED noted as True, proceeding with scan.
11:11:05.177: [APP/PROC/WEB.0] [2022-08-04 15:11:05,176 DEBUG clients.py::__init__:L37 :  Set clamav endpoint_url as 'http://tanf-dev-clamav-rest.apps.internal:9000/scan'
11:11:05.177: [APP/PROC/WEB.0] Set clamav endpoint_url as 'http://tanf-dev-clamav-rest.apps.internal:9000/scan'
11:11:05.177: [APP/PROC/WEB.0] [2022-08-04 15:11:05,176 DEBUG clients.py::scan_file:L64 :  Initiating virus scan for file: Section1unsafe.txt
11:11:05.177: [APP/PROC/WEB.0] Initiating virus scan for file: Section1unsafe.txt
11:11:05.184: [APP/PROC/WEB.0] [2022-08-04 15:11:05,183 DEBUG clients.py::scan_file:L90 :  File scan marked as INFECTED for file: Section1unsafe.txt
11:11:05.184: [APP/PROC/WEB.0] File scan marked as INFECTED for file: Section1unsafe.txt
11:11:05.192: [APP/PROC/WEB.0] WARNING Bad Request: /v1/data_files/
11:
```

**Comment by n0remac** [comment]
Andrew and I tested this again on local and it worked. Shipping to Prod will be the only way to test there.

**Review by andrew-jameson — APPROVED** [review]

**Pull request merged**